_pages/fsa/dynamicforms/json-structure.md

```diff
@@ -0,0 +1,472 @@
+  formGroups: DynamicFormGroup[];
+  formId: string;
+  cssClass?: string;
+}
+export interface DynamicFormGroup {
+  groupCode?: string;
+  fieldConfigs: FieldConfig[];
+  cssClass?: string;
+  dependsOn?: DependsOn;
+}
+
+export interface FieldConfig {
+  name?: string;
+  required?: boolean;
+  disabled?: boolean;
+  label?: LocalizedString;
+  options?: FieldOption[];
+  cssClass?: string;
+  gridClass?: string;
+  apiValue?: ApiConfig;
+  placeholder?: string;
+  fieldType: string;
+  value?: any;
+  hidden?: boolean;
+  error?: LocalizedString;
+  validations?: ValidatorFunction[];
+  dependsOn?: DependsOn;
+  prefillValue?: PrefillValue;
+  readonly?: boolean;
+  maxUploads?: number;
+  accept?: string[];
+  maxFileSize?: number;
+  multiple?: boolean;
+}
+export interface DependsOn {
+  hide?: boolean;
+  controls?: ControlDependency[];
+}
+export interface ApiConfig {
+  url: string;
+  param?: string;
+}
+
+export interface PrefillValue {
+  targetObject: string;
+  targetValue?: string;
+}
+
+export interface ControlDependency {
+  controlName?: string;
+  conditions?: ValidatorFunction[];
+}
+
+export interface LocalizedString {
+  default?: string;
+  [lang: string]: string;
+}
+
+export interface ValidatorFunction {
+  name: string;
+  arguments?: ValidatorArgument[];
+}
+
+export interface ValidatorArgument {
+  value: string;
+}
+
+export interface FieldOption {
+  name: string;
+  label: LocalizedString;
+  selected?: boolean;
+  icon?: string;
+}
+
+```
+Now we will try to explain interface structure in more details:
+
+## FormDefinition 
+Represents top level entity which hold group of form field
+- [formGroups](#dynamicformgroup) - Defines array of form groups
+- formId - Attaches ID to the form
+- cssClass - Enables adding unique CSS class to the form wrapper. More info can be found [Here]({{ site.baseurl }}{% link _pages/fsa/dynamicforms/styling.md %})
+
+## DynamicFormGroup
+Represents group of form controls
+- groupCode - Adds name to the form group which can be found in payload on form submit. In this example of submitted form data object, we can see that form controls are wrapped in two form groups:
+```typescript
+groupOne: 
+    control1: "0"
+groupTwo: 
+    control2: "2020-12-31"
+    control3: "YEARLY"
+```
+- fieldConfigs - Array of controls in this form group
+- cssClass - Enables adding unique CSS class to the form group wrapper. More info can be found [Here]({{ site.baseurl }}{% link _pages/fsa/dynamicforms/styling.md %})
+- dependsOn - With this property, we can define whole group`s visibility by defining control name on which group depends on.
+
+## FieldConfig
+Represents individual control
+- name - Sets control name
+- required - Defines if control is required
+- disabled - Defines if control is disabled ( Disabled fields are not picked up on submit )
+- [label](#localizedstring) - Defines control label ( supports localization)
+- options - Defines select`s, radio buttons etc.. options
```

Comment:
Defines available options for the following input types: dropdown, radio button, checkbox